backend/src/db/migrations/44-remove-waiting-events-no-null-tracks-constraint.sql

```diff
@@ -0,0 +1,6 @@
+-- There is already the `ready_event_has_fields` constraint in
+-- 05-events.sql. The `no_null_tracks` constraint was added later
+-- to also prevent `waiting` events from having no tracks, but
+-- it turns out that we actually don't want that constraint for
+-- the purpose of creating dummy events for waiting.
+alter table all_events drop constraint no_null_tracks;
```

Comment:
I think our constraint was just buggy. We still want a constraint to make sure there are no `null` elements inside the array. The check is just buggy because `array_position(tracks, null)` also returns `null` if the array is empty. So the constraint should have been:

```sql
add constraint no_null_tracks
    check (array_length(tracks, 1) = 0 or array_position(tracks, null) is null)
```